superset/db_engine_specs.py

```diff
@@ -1003,11 +1003,15 @@ def fetch_result_sets(cls, db, datasource_type):
 
     @classmethod
     def fetch_data(cls, cursor, limit):
+        import pyhive
         from TCLIService import ttypes
         state = cursor.poll()
         if state.operationState == ttypes.TOperationState.ERROR_STATE:
             raise Exception('Query error', state.errorMessage)
-        return super(HiveEngineSpec, cls).fetch_data(cursor, limit)
+        try:
+            return super(HiveEngineSpec, cls).fetch_data(cursor, limit)
+        except pyhive.exc.ProgrammingError:
```

Comment:
This seems dangerous as in theory it's the dbapi error that should be returned if there's anything wrong with the SQL... https://www.python.org/dev/peps/pep-0249/#programmingerror

"Table not found" should not succeed with an empty result set...

Reply:
@mistercrunch, if there's something wrong with the query it would raise the exception when calling `.execute()`, no? It seems to me that catching exceptions in `.fetchall()` should be fine, since the query has already ran.

Reply:
If that's the case, I'm fine with the solution here, but we should have a comment explaining all this here.